developer/reference/articles/tooling/blade-cli/08-extending-blade-cli/02-creating-custom-commands-for-blade-cli.markdown

```diff
@@ -0,0 +1,102 @@
+---
+header-id: creating-custom-commands-for-blade-cli
+---
+
+# Creating Custom Commands for Blade CLI
+
+[TOC levels=1-4]
+
+To create a custom command for Blade CLI, follow these steps:
+
+| **Note:** This article creates a Gradle-based command project. These steps can
+| be completed for a Maven-based project too.
+
+1.  [Create a generic OSGi module](/docs/7-2/reference/-/knowledge_base/r/creating-a-project).
+
+2.  You'll leverage [JCommander](http://jcommander.org/) and the Blade CLI API
+    to create your custom command. Add these dependencies in your build file.
+    For example, a `build.gradle` file's `dependencies` block looks like this:
+
+    ```groovy
+    dependencies {
+        compileOnly group: "com.beust", name: "jcommander", version: "1.72"
+        compileOnly group: "com.liferay.blade", name: "com.liferay.blade.cli", version: "latest.release"
+    }
+    ```
+
+3.  Build a Command class by extending the
+    [`BaseCommand`](https://github.com/liferay/liferay-blade-cli/blob/master/cli/src/main/java/com/liferay/blade/cli/command/BaseCommand.java)
```

Comment:
I typically try to avoid fully qualifying the class name within the text, as it makes the instructions more bloated (see I included the link and import statement below for reference). Since it is a Liferay-specific class, though, perhaps this would be a case where it'd be more useful. I can update it.

Reply:
Yeah true, they can just go to the link and look at it for the FQN.